plugins/out_es/es.c

```diff
@@ -159,14 +163,53 @@ static inline int es_pack_map_content(msgpack_packer *tmp_pck,
             msgpack_pack_map(tmp_pck, v->via.map.size);
             es_pack_map_content(tmp_pck, *v, ctx);
         }
+        /*
+         * The value can be any data type, if it's an array we need to
+         * pass it to es_pack_array_content.
+         */
+        else if (v->type == MSGPACK_OBJECT_ARRAY)
+        {
+          msgpack_pack_array(tmp_pck, v->via.array.size);
+          es_pack_array_content(tmp_pck, *v, ctx);
+        }
         else {
             msgpack_pack_object(tmp_pck, *v);
         }
     }
-
     return 0;
 }
 
+/*
+  * Iterate through the array and sanitize elements.
+  * Mutual recursion with es_pack_map_content.
+  */
+static int es_pack_array_content(msgpack_packer *tmp_pck,
+                                 msgpack_object array,
+                                 struct flb_elasticsearch *ctx)
+{
+  msgpack_object *e;
```

Comment:
indentation must be 4 spaces: https://github.com/fluent/fluent-bit/blob/master/CONTRIBUTING.md#coding-style